### ISSUE TITLE
Dead keys not working in a lot of applications

### ISSUE THREAD

**Issue by terminalnode** [swaywm/sway#4146]
* Sway Version: sway version 1.1-rc1-5-g18ce0eec (May 13 2019, branch 'master')
* Debug Log: [sway.log](https://termbin.com/dg5o)
* Configuration File: [config](https://gitlab.com/terminalnode/dotfiles/raw/6f0a83608f151efdaa2c86561d9cceb3d1886569/.config/sway/config)

As the title states, dead keys are not working in a lot of applications. I haven't been able to deduce which applications it works in and in which it doesn't, at first it seemed to be xwayland applications where it works and native applications where it does not but that theory doesn't seem to hold water as many applications handle it fine regardless if run as wayland or xwayland (firefox for example).

Running xev my dead `, ´ and ^ all register correctly as dead_[keyname].

I'm not sure when it started, but I didn't notice it before upgrading from stable sway to rc1 and thus suspect it might have something to do with that. I'm fairly certain this hasn't always been the case, but since I don't use circumflexes and accents a lot it's possible that I just haven't noticed.

I have tried removing my xkb_options and dual-keyboard layout configuration in sway to no avail.

Here are some applications I've tested in, which is all graphical applications I could think of.

**Works:**
discord, xwayland firefox, `GDK_BACKEND=wayland firefox`, xterm, calibre, qutebrowser, bitwarden, steam, anki, rofi

**Does not work:**
spotify, termite, krita, arduino IDE, netbeans, lutris, libreoffice

**Comment by terminalnode** [comment]
If someone has a configuration where dead keys are working in the applications above I'd be happy to try it out to rule out any oddities in my configuration and possibly find out what part of the configuration is causing the issue.

**Comment by progandy** [comment]
xkb_layout se(dvorak_a5),se

shouldn't that be two lines? If it doesn't work maybe try to add "basic" after the comma for the variant.

    xkb_layout se,se
    xkb_variant dvorak_a5,

**Comment by terminalnode** [comment]
That (without basic) seems to have fixed it! I got the syntax used before (`se(dvorak_a5),se`) from some other config and thought it looked cleaner. Tested it out and didn't notice any issues until way later, so I didn't connect the issues with the new syntax.

I wonder why the old configuration worked fine in some applications but not in others though, that's the part that's the most confusing to me.

Is this confirmed not a sway issue then, or could it be related to how sway parses these options?
